gradle-nonnull-plugin/plugin/src/main/groovy/com.novoda.gradle.nonnull/GeneratePackageAnnotationsTask.groovy

```diff
@@ -0,0 +1,71 @@
+package com.novoda.gradle.nonnull
+
+import org.gradle.api.DefaultTask
+import org.gradle.api.file.FileVisitDetails
+import org.gradle.api.tasks.TaskAction
+
+class GeneratePackageAnnotationsTask extends DefaultTask {
+
+    def outputDir
+
+    def variant
+
+    @TaskAction
+    void generatePackageAnnotations() {
+        description = "Generate package-info.java classes"
```

Comment:
This could be a bit more specific, imho. 
May be `Annotates the source packages with @ParametersAreNonnullByDefault` ?